code/modules/language/language.dm

```diff
@@ -5,6 +5,11 @@
 */
 
 /datum/language
+	/// abstract type
+	var/abstract_type = /datum/language
+	/// uid
+	var/id
+	// TODO: ref languages by id in code, so we can rename as needed
```

Comment:
yes im reworking species a bit because god fucking damnit this was obnoxious to deal with